src/common/viz/FigureExtractor.js

```diff
@@ -17,210 +17,334 @@ define(['./Utils'], function (Utils) {
+                return [];
+            }
+        }
+
+        getGraphNode (node) {
+            return this._getContainmentParentNodeAt(node, 'Graph');
+        }
+
+        _getContainmentParentNodeAt (node, metaType) {
+            let currentNode = node,
+                parentId = currentNode.getParentId();
+            const isMetaType = node => this.getMetaType(node) === metaType;
+            while (parentId !== null && !isMetaType(currentNode)) {
+                currentNode = this._client.getNode(parentId);
+                parentId = currentNode.getParentId();
+            }
+            return isMetaType(currentNode) ? currentNode : null;
+        }
+
+        getMetaType (node) {
+            const metaTypeId = node.getMetaTypeId();
+            const metaNode = this._client.getNode(metaTypeId);
+            if(metaNode) {
+                return metaNode.getAttribute('name');
+            }
+        }
+
+        GMENodeToMetadataJSON (node, shallow=false) {
+            const json = {
+                id: node.getId(),
+                parent_id: node.getParentId(),
+                execution_id: this.getExecutionId(node),
+                meta_type: this.getMetaType(node),
+                attributes: {},
+                attributes_meta: {}
+            };
+
+            node.getOwnAttributeNames().forEach(name => {
+                json.attributes[name] = node.getAttribute(name);
+            });
 
-        if(metadataBaseNode) {
-            return node.getChildrenIds().filter(id => {
-                return this._client.isTypeOf(id, metadataBaseNode.getId());
+            node.getOwnValidAttributeNames().forEach(name => {
+                json.attributes_meta[name] = node.getAttribute(name);
             });
-        } else {
-            return [];
+
+            if(!shallow) {
+                json.children = [];
+                const children = this.getMetadataChildrenIds(node).map(id => this._client.getNode(id));
+                children.forEach(node => {
+                    json.children.push(this.GMENodeToMetadataJSON(node));
+                });
+            }
+            return json;
         }
-    };
+    }
 
-    FigureExtractor.prototype.getGraphNode = function(node) {
-        return this._getContainmentParentNodeAt(node, 'Graph');
-    };
+    class CoreFigureExtractor extends AbstractFigureExtractor {
+        constructor(core, rootNode) {
+            super();
+            this._core = core;
+            this._rootNode = rootNode;
+        }
 
-    FigureExtractor.prototype._getContainmentParentNodeAt = function(node, metaType){
-        let currentNode = node,
-            parentId = currentNode.getParentId();
-        const isMetaType = node => this.getMetaType(node) === metaType;
-        while (parentId !== null && !isMetaType(currentNode)) {
-            currentNode = this._client.getNode(parentId);
-            parentId = currentNode.getParentId();
+        getExecutionId (node) {
```

Comment:
If you remove `execution_id` from the JSON representation, this could be defined in the base class and operate on the JSON representation rather than GME nodes.